num/src/bigint.rs

```diff
@@ -1060,16 +1081,230 @@ macro_rules! construct_bigint {
     };
 }
 
+macro_rules! construct_signed_bigint_methods {
+    ( $ name: ident, $ n_words: expr ) => {
+        impl From<i8> for $name {
+            fn from(init: i8) -> $name {
+                let bytes = init.to_le_bytes();
+                let mut ret = [if init.is_negative() {
+                    ::core::u8::MAX
+                } else {
+                    0
+                }; $n_words * 8];
+                for i in 0..bytes.len() {
+                    ret[i] = bytes[i]
+                }
+                $name::from_le_bytes(ret)
+            }
+        }
+
+        impl From<i16> for $name {
+            fn from(init: i16) -> $name {
+                let bytes = init.to_le_bytes();
+                let mut ret = [if init.is_negative() {
+                    ::core::u8::MAX
+                } else {
+                    0
+                }; $n_words * 8];
+                for i in 0..bytes.len() {
+                    ret[i] = bytes[i]
+                }
+                $name::from_le_bytes(ret)
+            }
+        }
+
+        impl From<i32> for $name {
+            fn from(init: i32) -> $name {
+                let bytes = init.to_le_bytes();
+                let mut ret = [if init.is_negative() {
+                    ::core::u8::MAX
+                } else {
+                    0
+                }; $n_words * 8];
+                for i in 0..bytes.len() {
+                    ret[i] = bytes[i]
+                }
+                $name::from_le_bytes(ret)
+            }
+        }
+
+        impl From<i64> for $name {
+            fn from(init: i64) -> $name {
+                let bytes = init.to_le_bytes();
+                let mut ret = [if init.is_negative() {
+                    ::core::u8::MAX
+                } else {
+                    0
+                }; $n_words * 8];
+                for i in 0..bytes.len() {
+                    ret[i] = bytes[i]
+                }
+                $name::from_le_bytes(ret)
+            }
+        }
+
+        impl From<i128> for $name {
+            fn from(init: i128) -> $name {
+                let bytes = init.to_le_bytes();
+                let mut ret = [if init.is_negative() {
+                    ::core::u8::MAX
+                } else {
+                    0
+                }; $n_words * 8];
+                for i in 0..bytes.len() {
+                    ret[i] = bytes[i]
+                }
+                $name::from_le_bytes(ret)
+            }
+        }
+
+        impl $name {
+            /// Minimum value
+            pub const MIN: $name = {
+                let mut min = [0u64; $n_words];
+                min[$n_words - 1] = 0x8000_0000_0000_0000;
+                $name(min)
+            };
+
+            /// Maximum value
+            pub const MAX: $name = {
```

Comment:
Oh, sorry, probably the removal of `MAX`/`MIN` was not a breaking change. Can you pls make sure that we are not breaking API of existing bignum types?

Reply:
@dr-orlovsky We have no breaking API changes in this PR. I am just moving some methods to `construct_signed_bigint_methods` macro, because they need different implementation on `signed` and `unsigned`. (e.g. max value is 0xffff.. in unsigned and 0x7fff... in signed) So we can just `3.11.0`, not `4.0.0`